hal/src/sercom/v2/uart/pads_thumbv6m.rs

```diff
@@ -264,7 +264,7 @@ where
     }
 }
 
-#[cfg(feature = "samd21")]
+#[cfg(feature = "samd2x")]
```

Comment:
RTS/CTS pads don't exist for SAMD20. I think this chip should get special cases for the implementation of `ValidPads`, and disable the current implementations, so that `RTS` and `CTS` are both `NoneT`. This would be simpler that changing the `impl_rxpotxpo` macro, though it could be confusing the the end user when creating a new `Pads` type.

There are also slight changes int `USART.CTRLA.TXPO` field. We need to make sure we never write `0x2` to this field for SAMD20 targets.

 In any case, whatever differences between the SAMD20 and the other chips definitely need to be documented in the relevant module-level documentations.

Paging @bradleyharden - What do you think of all this?